internal/driver/driver.go

```diff
@@ -78,14 +88,22 @@ func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModel.As
 	return nil
 }
 
+var (
+	GetEndpoints = opcua.GetEndpoints
+)
+
+var (
+	SelectEndPoint = opcua.SelectEndpoint
+)
+
 // creates the options to connect with a opcua Client based on the configured options.
 func (d *Driver) createClientOptions() ([]opcua.Option, error) {
-	availableServerEndpoints, err := opcua.GetEndpoints(d.serviceConfig.OPCUAServer.Endpoint)
+	availableServerEndpoints, err := GetEndpoints(d.serviceConfig.OPCUAServer.Endpoint)
 	if err != nil {
 		d.Logger.Error("OPC GetEndpoints: %w", err)
 		return nil, err
 	}
-	credentials, err := getCredentials(d.serviceConfig.OPCUAServer.CredentialsPath)
+	credentials, err := d.getCredentials(d.serviceConfig.OPCUAServer.CredentialsPath)
```

Comment:
I would change the parameter from "d" to something meaningful such as driver etc.

Reply:
I think that could rather be done as a global refacoring since this seems to be the edgex convention, or at least of the guys who wrote this base device service. So the driver variable is calles d verywhere and changing it only here would make no sense. Or did you mean I should change it in the whole project, not just here? :D

Reply:
Reasonable :D